src/trace/implementations/mod.rs

```diff
@@ -510,7 +500,7 @@ pub mod containers {
     use crate::trace::IntoOwned;
 
     /// A general-purpose container resembling `Vec<T>`.
-    pub trait BatchContainer: 'static {
+    pub trait BatchContainer: for<'a> PushInto<Self::ReadItem<'a>> + 'static {
```

Comment:
Discussed in person: We could opt out of requiring `BatchContainer` to be `PushInto<...>` which would permit implementing containers that cannot be pushed at. For the moment, the change as implemented is probably the right thing to do, but in the future we could split the trait into `ReadBatchContainer` and `BatchContainer` where the former does not expose any functionality to modify the contents of the container. There's a similar idea in flatcontainer: https://github.com/antiguru/flatcontainer/pull/27